### PR TITLE
1251 testcase sub-identities

### LINKED ISSUE #1251: testcase: subidentities

### PR DESCRIPTION
**Pull request by Ekaanth**

### CODE REVIEW

File: mocks/handlers/accountHandler.js [hunk 1]
```diff
@@ -48,6 +48,7 @@ export const accountHandler = graphql.query('getAccount', (req, res, ctx) => {
           feeFrozen: '198480993716691',
           formattedFeeFrozen: '19,848.0993 DOT',
         },
+        subAccounts: [],
```
Comment (ishrak-imam) [inline mocks/handlers/accountHandler.js]: Don't we need any mock data for the sub accounts??

**Review by ishrak-imam — APPROVED** [review]

File: src/ui/screens/SubIdentities/AddSubIdentity.test.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+const bottomSheetModal = {
+  dismiss: jest.fn(),
+};
+
+const addSubIdentity = {
+  subIdentity: jest.fn(),
+};
+
+describe('', () => {
```
Comment (silva-fj) [inline src/ui/screens/SubIdentities/AddSubIdentity.test.tsx]: Let's add the description

File: src/ui/screens/SubIdentities/AddSubIdentity.test.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import {render, fireEvent} from 'src/testUtils';
+import {AddSubIdentity} from './AddSubIdentity';
+
+jest.useFakeTimers();
```
Comment (silva-fj) [inline src/ui/screens/SubIdentities/AddSubIdentity.test.tsx]: Why do you need this?

File: src/ui/screens/SubIdentities/AddSubIdentity.test.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    await findByText('Add Identity');
+  });
+
+  it('should close the bottom-sheets when pressed on cancel button', async () => {
+    const bottomSheetsSpy = jest.spyOn(bottomSheetModal, 'dismiss');
+    const {findByText, findByTestId} = render(
+      <AddSubIdentity onClose={bottomSheetModal.dismiss} onAddPress={addSubIdentity.subIdentity} />,
+    );
+    await findByText('Cancel');
+    fireEvent.press(await findByTestId('cancel-identity-button'));
```
Comment (silva-fj) [inline src/ui/screens/SubIdentities/AddSubIdentity.test.tsx]: In this case (finding the button to fire a press event) it is better to get it by role or text, the `testID` is needed when you want to assert that the button is disabled or not.

File: src/ui/screens/SubIdentities/AddSubIdentity.test.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    expect(bottomSheetsSpy).toBeCalledTimes(1);
+  });
+
+  it('should add new sub-identity', async () => {
```
Comment (silva-fj) [inline src/ui/screens/SubIdentities/AddSubIdentity.test.tsx]: How about:
` it('should call onAddPress with the right arguments when the form is valid and Add Identity button is pressed', () => {...})`

We are not actually testing that the sub-identity was added to something

File: src/ui/screens/SubIdentities/RegisterSubIdentitiesScreen.test.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import {NavigationProp, RouteProp} from '@react-navigation/native';
+import {AccountsStackParamList} from '@ui/navigation/navigation';
+import {render} from 'src/testUtils';
+import {RegisterSubIdentitiesScreen} from './RegisterSubIdentitiesScreen';
+import {registerSubIdentitiesScreen} from '@ui/navigation/routeKeys';
+
+jest.useFakeTimers();
```
Comment (silva-fj) [inline src/ui/screens/SubIdentities/RegisterSubIdentitiesScreen.test.tsx]: Same, why do you need to add this?

File: src/ui/screens/SubIdentities/AddSubIdentity.test.tsx (hunk 3 above)
Comment (Ekaanth) [reply src/ui/screens/SubIdentities/AddSubIdentity.test.tsx]: Initially, when I was writing a test case using the `test` function, I had an import teardown, so adding this resolved it. Now I'm testing within describe, so I don't need it. It was removed in the latest push.

File: src/ui/screens/SubIdentities/AddSubIdentity.test.tsx (hunk 5 above)
Comment (Ekaanth) [reply src/ui/screens/SubIdentities/AddSubIdentity.test.tsx]: > We are not testing that the sub-identity was added to something

In this component, we don't have the actual implementation. While this component accepts the `address` and sends an object, that's what I'm mocking in this.

File: mocks/handlers/accountHandler.js (hunk 1 above)
Comment (Ekaanth) [reply mocks/handlers/accountHandler.js]: I have tried to mock the data, but I think the data I mocked is wrong, and unable to find a relevant example account. Is it possible for you to provide a mock for this?

Comment (ishrak-imam) [reply mocks/handlers/accountHandler.js]:
```
{
  substrateChainAccount(address: "13KJ3t8w1CKMkXCmZ6s3VwdWo4h747kXE88ZNh6rCBTvojmM") {
    address
    subAccounts {
      address
    }
  }
}
```

This account has sub accounts. You can get the mock from here. Add it as an external account in the app and you can get the data.

**Review by silva-fj — APPROVED** [review]

**Pull request merged**